src/ducks/wallet/trezor.ts

```diff
@@ -0,0 +1,96 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+// @types/trezor-connect doesn't have .stellarGetAddress()
+// @ts-ignore
+import TrezorConnect from "trezor-connect";
+
+export const fetchTrezorStellarAddressAction = createAsyncThunk<
+  // Return type of the payload creator
+  string,
+  // First argument to the payload creator
+  undefined,
+  // Types for ThunkAPI
+  { rejectValue: RejectMessage }
+>(
+  "wallet/trezor/fetchTrezorStellarAddressAction",
+  async (_, { rejectWithValue }) => {
+    try {
+      const trezorResponse = await TrezorConnect.stellarGetAddress({
+        path: "m/44'/148'/0'",
+      });
+
+      if (!trezorResponse.success) {
+        const errorMessage = trezorResponse.payload?.error
+          ? `Couldn't connect to your Trezor device: ${trezorResponse.payload?.error}`
+          : "Couldn't connect to your Trezor device, please try again.";
+
+        return rejectWithValue({
+          errorMessage,
+        });
+      }
+
+      return trezorResponse.payload.address;
+    } catch (error) {
+      return rejectWithValue({
+        errorMessage: error.toString(),
+      });
+    }
+  },
+);
+
+interface RejectMessage {
+  errorMessage: string;
+}
+
+export enum ActionStatus {
+  PENDING = "PENDING",
+  SUCCESS = "SUCCESS",
+  ERROR = "ERROR",
+}
+
+interface InitialState {
+  data: string | null;
+  status: ActionStatus | undefined;
+  errorMessage?: string;
+}
+
+const initialState: InitialState = {
+  data: null,
+  status: undefined,
+  errorMessage: undefined,
+};
+
+const walletTrezorSlice = createSlice({
+  name: "account",
```

Comment:
is the plan to name all the sign-in slices with "account"? It might be confusing understanding the dispatch logs

eg. "account/reset" it would be hard to tell if that was an account from secret key login or from Trezor/Ledger etc.

Thoughts on making the name "accountTrezor" or something like that?

Reply:
Thanks! I missed that one. I think it should be `walletTrezor` (directory/filename) to follow the structure. Are you OK with that?

Reply:
Works for me! 👍